### ISSUE TITLE
Rule Change: Extend no-empty-definitions to cover empty footnote definitions

### ISSUE THREAD

**Issue by TKDev7** [eslint/markdown#435]
### What rule do you want to change?

no-empty-definitions

### What change do you want to make?

Generate more warnings

### How do you think the change should be implemented?

A new default behavior

### Example code

```markdown
<!-- should be reported as empty -->
[^note]:
```

### What does the rule currently do for this code?

Does not report `[^note]:` as empty.

### What will the rule do after it's changed?

Reports `[^note]:` as empty footnote definition.

### Participation

- [x] I am willing to submit a pull request to implement this change.

**Comment by nzakas** [comment]
It looks like `no-duplicate-definitions` applies to both definitions and footnote definitions, so to keep things consistent it seems like we should add this as an option to `no-empty-definitions` instead of having a different rule.

Marking as accepted.
